### ISSUE TITLE
Define a Bicycle constructor comparison incorrect

### ISSUE THREAD

**Issue by electricbolt** [dart-lang/site-www#3196]
Page: https://codelabs.developers.google.com/codelabs/from-java-to-dart/#1

I'm using DartPad to assemble the code as per the instructions. Under the section:

**Define a Bicycle constructor**
> Add the following constructor to the Bicycle class:

`Bicycle(this.cadence, this.speed, this.gear);`

Under the Observations section it says the code above is equivalent to the following:

```
Bicycle(int cadence, int speed, int gear) {
  this.cadence = cadence;
  this.speed = speed;
  this.gear = gear;
}
```

This doesn't appear to be the case. Using the multi-line constructor the DartPad will issue the following errors:

```
Error compiling to JavaScript:
Warning: Interpreting this as package URI, 'package:dartpad_sample/main.dart'.
lib/main.dart:2:7:
Error: Field 'cadence' should be initialized because its type 'int' doesn't allow null.
  int cadence;
      ^^^^^^^
lib/main.dart:3:7:
Error: Field 'speed' should be initialized because its type 'int' doesn't allow null.
  int speed;
      ^^^^^
lib/main.dart:4:7:
Error: Field 'gear' should be initialized because its type 'int' doesn't allow null.
  int gear;
      ^^^^
Error: Compilation failed.
```

Sorry, I don't know enough about Dart yet to understand why. But the documentation is clearly wrong now.

**Comment by parlough** [comment]
Hey @electricbolt, thanks so much for the report!

With the introduction of [sound null safety](https://dart.dev/null-safety) these are no longer the same, as for Dart to know the three fields are not ever null(due to having the `int` type, not `int?`), they need to be assigned with `this` constructor parameters or in the constructor initializer list. 

So the equivalent in this case(using an initializer list) would be:

```dart
  Bicycle(int cadence, int speed, int gear)
      : this.cadence = cadence,
        this.speed = speed,
        this.gear = gear;
```

If the fields were marked as nullable(int?) or [`late`](https://dart.dev/null-safety/understanding-null-safety#late-variables) though, then the original example constructor would technically work as well, but their recommended constructor or the initializer list are best.

Thanks again, we'll definitely work on getting this equivalent suggestion fixed!

**Comment by kwalrath** [comment]
Unassigned myself since I don't know the new codelab publishing process. I'm willing, but I'd need help from @domesticmouse.

**Comment by domesticmouse** [comment]
@kwalrath I've pinged you on the gDoc that is the source for this codelab. Let me know when you have made the appropriate changes and I'll get the publishing process under way.

**Comment by kwalrath** [comment]
OK, I've fixed this issue and #3223. Thanks for taking care of the publishing process!

**Comment by kwalrath** [comment]
Verified that this fix is published.
